src/bundle.c

```diff
@@ -1412,7 +1412,7 @@ gboolean check_bundle(const gchar *bundlename, RaucBundle **bundle, gboolean ver
 	if (is_remote_scheme(bundlescheme)) {
 #if ENABLE_NETWORK
 		ibundle->origpath = g_strdup(bundlename);
-		ibundle->path = g_build_filename(g_get_tmp_dir(), "_download.raucb", NULL);
+		ibundle->path = g_build_filename(g_dir_make_tmp("rauc-XXXXXX", NULL), "download.raucb", NULL);
```

Comment:
Leaks.

Reply:
And while at it, you may want to check for `NULL` and return, even if the error is unlikely.